FlightScheduler/APICall.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  APICall.swift
+//  FlightScheduler
+//
+//  Created by macbook on 26/02/2020.
+//  Copyright © 2020 Practice. All rights reserved.
+//
+
+import Foundation
+
+enum APIError: Error {
+    case responseProblem
+    case decodingProblem
+    case encodingProblem
+
+}
+
+let GRANT_TYPE = "credentials"
+let CLIENT_ID = "xuzza9nuseytv3m59fyytrue"
+let CLIENT_SECRET = "mCH385mArn"
+
+struct APIRequest {
+    let resourceURL: URL
+
+    init(endpoint: String) {
```

Comment:
It's not good practice to initialize an endpoint like you have done here. The way you've done it here, this class is very limited and can only be used  with one url.

Reply:
Changes made